projects/ssr-tests/src/ssr-testing_spec.ts

```diff
@@ -0,0 +1,85 @@
+import * as Log from './log.utils';
+import * as ProxyServer from './proxy.utils';
+import * as Ssr from './ssr.utils';
+
+const BACKEND_BASE_URL: string = process.env.CX_BASE_URL || '';
+
+describe('SSR E2E', () => {
+  let proxy: any;
+  const REQUEST_PATH = '/electronics-spa/en/USD/';
+
+  beforeAll(async () => {
+    Log.clearSsrLogFile();
+    await Ssr.startSsrServer();
+  });
+
+  afterEach(async () => {
+    await proxy.close();
+  });
+
+  afterAll(async () => {
+    await Ssr.killSsrServer();
+  });
+
+  it('should receive success response with request', async () => {
+    proxy = await ProxyServer.startProxyServer({
+      target: BACKEND_BASE_URL,
+    });
+    const response: any = await ProxyServer.sendRequest(REQUEST_PATH);
+    expect(response.statusCode).toEqual(200);
+
+    // Rendering should not complete in the first request.
+    // App should fall back to csr.
+    Log.assertMessages([
+      `Rendering started (${REQUEST_PATH})`,
+      `Request is waiting for the SSR rendering to complete (${REQUEST_PATH})`,
+    ]);
+  });
+
+  // Wait for SSR server to complete rendering.
+  it('should complete rendering within a minute', async () => {
+    await Log.waitUntilLogContainsText(`Rendering completed (${REQUEST_PATH})`);
+    Log.assertMessages([`Rendering completed (${REQUEST_PATH})`]);
+  });
+
+  xit('should receive cached response with next request', async () => {
+    proxy = await ProxyServer.startProxyServer({
+      target: BACKEND_BASE_URL,
+    });
+    const response: any = await ProxyServer.sendRequest(REQUEST_PATH);
+    expect(response.statusCode).toEqual(200);
+
+    Log.assertMessages(['Render from cache (/)']);
+  });
+
+  xit('should receive 404 response when page is not existing', () => {});
+
+  xit('should receive 500 error response with request', async () => {
+    proxy = await ProxyServer.startProxyServer({
+      target: BACKEND_BASE_URL,
+      throwStatus: 500,
+    });
+    proxy.on('proxyRes', function (proxyRes: any) {
+      proxyRes.statusCode = 500;
+    });
+    const response: any = await ProxyServer.sendRequest('/');
+    expect(response.statusCode).toEqual(500);
+  });
+
+  // Note: Currently, the ssr server still responds with 200 quickly despite the proxy delay
+  xit('should receive 500 error response with timed-out request', async () => {
```

Comment:
[MEDIUM]
What about leaving `//TODO` comments before each `xit()`, so SonarQube won't let us accidentally merge the Epic branch "SSR error handling" to `develop` until we write those tests. Does it make sense?